### PR TITLE
[MXNET-892] ONNX export/import: DepthToSpace, SpaceToDepth operators

### PR DESCRIPTION
**Pull request by vandanavk**
## Description ##
Adding DepthToSpace, SpaceToDepth operators to ONNX import/export

v2: Added tests for SpaceToDepth. 
v3: Updated the test for SpaceToDepth - removed hard coded input/output. Since ONNX export test executed import, export, import, maintaining the custom test only in mxnet_export_test.py.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Added DepthToSpace, SpaceToDepth operator import/export
- Added tests for DepthToSpace

## Comments ##
Temporarily adding tests for SpaceToDepth in mxnet_export_test.py and onnx_import_test.py.

### CODE REVIEW

**Comment by vrakesh** [comment]
@mxnet-label-bot [pr-awaiting-review]

**Comment by vandanavk** [comment]
@anirudhacharya @zhreshold @Roshrini for review

File: tests/python-pytest/onnx/import/test_cases.py [hunk 1]
```diff
@@ -85,7 +85,8 @@
     'test_operator_exp',
     'test_operator_maxpool',
     'test_operator_params',
-    'test_operator_permute2'
+    'test_operator_permute2',
+    'test_depthtospace'
```
Comment (Roshrini) [inline tests/python-pytest/onnx/import/test_cases.py]: no test for SpaceToDepth in onnx?

Comment (vandanavk) [reply tests/python-pytest/onnx/import/test_cases.py]: The test is in review. Hasn't been merged into ONNX yet.

**Comment by sandeep-krishnamurthy** [comment]
@Roshrini @anirudhacharya - Can you please take a look at this PR?

File: tests/python-pytest/onnx/import/test_cases.py (hunk 1 above)
Comment (vandanavk) [reply tests/python-pytest/onnx/import/test_cases.py]: Added tests for SpaceToDepth

**Comment by Roshrini** [comment]
Thanks for adding test Vandana. Lgtm

**Comment by vandanavk** [comment]
Updated the test for SpaceToDepth - removed hard coded input/output. Since ONNX export test executed import, export, import, maintaining the custom test only in mxnet_export_test.py.

**Comment by vandanavk** [comment]
Addressed all the review comments so far. Please have a look @zhreshold @anirudhacharya @Roshrini

**Comment by Roshrini** [comment]
@zhreshold Can you take a look and merge if it looks good?

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 2]
```diff
@@ -2077,3 +2077,53 @@ def convert_sqrt(node, **kwargs):
...
+        name=name,
+    )
+    return [node]
+
+@mx_op.register("space_to_depth")
+def convert_spacetodepth(node, **kwargs):
+    """Map MXNet's space_to_depth operator attributes to onnx's
+    SpaceToDepth operator and return the created node.
+    """
+    onnx = import_onnx_modules()
```
Comment (zhreshold) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: can you try to modularize line 2111 to 2115, which is duplicated in every op converter, e.g., 2086-2090. You may not complete it in this PR, but I think it's a good practice to remove manual repetitive work.

**Comment by zhreshold** [comment]
@vandanavk See comment, if you find it's more convenient to implement that later in a separate PR, I can merge this one now.

**Pull request merged**